### PR TITLE
fix: Claim tokens above 1 dollar only

### PR DESCRIPTION
**Pull request by wkitsin**
1. claimPartially with `getMultipleAccountInfo` and Idempotent way to create token accounts for partner and admin 
2. claimAll to use Idempotent way

### CODE REVIEW

File: packages/sdk/src/referral.ts [hunk 1]
```diff
@@ -687,4 +656,141 @@ export class ReferralProvider {
...
+    );
+
+    const claimInstructionParams = await Promise.all(
+      result.map(async (item) => {
+        const tokenAccountData = AccountLayout.decode(item.data);
+
+        const isToken2022 = item.owner.equals(TOKEN_2022_PROGRAM_ID);
+        const tokenProgramId = isToken2022
+          ? TOKEN_2022_PROGRAM_ID
+          : TOKEN_PROGRAM_ID;
```
Comment (hjchan) [inline packages/sdk/src/referral.ts:692]: tokenProgramId is the item.owner, no need to check here, just pass in item.owner will do

File: packages/dashboard/src/app/dashboard/page.tsx [hunk 2]
```diff
@@ -418,10 +444,20 @@ const DashboardHeader: React.FC<{
...
+                disabled={totalUnclaimed === 0}
+                loading={isClaiming}
+              >
+                Claim
               </Button>
             </div>
+
+            <div className="mt-1 flex justify-end text-right text-xs text-[#E8F9FF]/50">
+              <div className="max-w-[300px]">
+                Only tokens with value more than $1 with be claimed.
```
Comment (hjchan) [inline packages/dashboard/src/app/dashboard/page.tsx:458]: only "strict" tokens, ask them to claim the others using the sdk

File: packages/dashboard/src/app/dashboard/page.tsx [hunk 3]
```diff
@@ -361,20 +379,28 @@ const DashboardHeader: React.FC<{
...
-    await sendAllTransactions(txsCompiled);
-    queryClient.refetchQueries({
-      queryKey: ["tokens"],
-    });
+    try {
+      setIsClaiming(true);
+      const txsCompiled = await referralProvider.claimPartially({
+        payerPubKey: wallet.publicKey,
+        referralAccountPubKey: referralPubkey,
+        strategy: { type: "token-list", tokenList: "strict" },
```
Comment (hjchan) [inline packages/dashboard/src/app/dashboard/page.tsx:390]: strategy not used in claimPartially

File: packages/sdk/src/referral.ts (hunk 1 above)
Comment (wkitsin) [reply packages/sdk/src/referral.ts:692]: oh ya.

**Review by hjchan — APPROVED** [review]

**Comment by wkitsin** [comment]
will merge this once priority fee is completed too

**Pull request merged**